panos/network.py

```diff
@@ -2125,6 +2126,190 @@ def _setup(self):
         RedistributionProfileBase._setup(self)
 
 
+class Rip(VersionedPanObject):
+    """Rip
+
+    Add to a :class:`panos.network.VirtualRouter` instance.
+
+    Args:
+        enable (bool): Enable RIP
+        reject_default_route (bool): Reject default route
+        allow_redist_default_route (bool): Allow Redistribute Default Route
+        delete_intervals (int): Delete Intervals
+        expire_intervals (int): Expire Intervals
+        interval_seconds (int): Interval Seconds (sec)
+        update_intervals (int): Update Intervals
+    """
+
+    NAME = None
+    CHILDTYPES = (
+        "network.RipInterface",
+        "network.RipAuthProfile",
+        "network.RipExportRules",
+    )
+
+    def _setup(self):
+        # xpaths
+        self._xpaths.add_profile(value="/protocol/rip")
+
+        # params
+        params = []
+
+        params.append(
+            VersionedParamPath("enable", path="enable", default=True, vartype="yesno")
+        )
+        params.append(
+            VersionedParamPath(
+                "reject_default_route",
+                default=True,
+                vartype="yesno",
+            )
+        )
+        params.append(
+            VersionedParamPath(
+                "allow_redist_default_route",
+                path="allow-redist-default-route",
+                vartype="yesno",
+            )
+        )
+        params.append(
+            VersionedParamPath(
+                "delete_intervals", path="/timers/delete-intervals", vartype="int"
+            )
+        )
+        params.append(
+            VersionedParamPath(
+                "expire_intervals", path="/timers/expire-intervals", vartype="int"
+            )
+        )
+        params.append(
+            VersionedParamPath(
+                "interval_seconds", path="/timers/interval-seconds", vartype="int"
+            )
+        )
+        params.append(
+            VersionedParamPath(
+                "update_intervals", path="/timers/update-intervals", vartype="int"
+            )
+        )
+
+        self._params = tuple(params)
+
+
+class RipInterface(VersionedPanObject):
```

Comment:
This class is missing a lot of params.  There is also a `choice` here at path `default-route` which should have its own variable (see `objects.AddressObjects` for an example of how to handle choices, they need their own param).